src/core/utils/number/numberUtils.ts

```diff
@@ -0,0 +1,100 @@
+import {FormatNumberOptions, ParseNumberOptions} from "./numberTypes";
+
+function formatNumber({providedOptions}: FormatNumberOptions) {
+  const {locale, ...otherOptions} = providedOptions;
+  const options = {
+    style: "decimal",
+    ...otherOptions
+  };
+
+  let numberFormatter: {
+    format: (x: number | bigint) => string;
+  };
+
+  try {
+    numberFormatter = new Intl.NumberFormat(
+      locale || [navigator.language, "en-GB"],
+      options
+    );
+  } catch (error) {
+    numberFormatter = {
+      format(x: number | bigint) {
+        return x.toLocaleString(locale);
+      }
```

Comment:
Can we add a comment here to describe what kind of error we are expecting?
I think this `catch` block handles the cases where `NumberFormat` (or `navigator`) is not supported by the browser, if so, we can add something like:
`// NumberFormat or navigator is not supported, just use toLocaleString for formatting`

Reply:
Yes, more descriptive. Thank you!